src/plugins/intel_gpu/src/graph/non_zero.cpp

```diff
@@ -0,0 +1,119 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+///////////////////////////////////////////////////////////////////////////////////////////////////
+#include "non_zero_inst.h"
+#include "primitive_type_base.h"
+#include "intel_gpu/runtime/memory.hpp"
+#include "intel_gpu/runtime/error_handler.hpp"
+#include "intel_gpu/runtime/debug_configuration.hpp"
+#include "json_object.h"
+#include <string>
+
+namespace cldnn {
+
+// -----------------------------------------------
+// count_nonzero
+// -----------------------------------------------
+primitive_type_id count_nonzero::type_id() {
+    static primitive_type_base<count_nonzero> instance;
+    return &instance;
+}
+
+layout count_nonzero_inst::calc_output_layout(count_nonzero_node const& node) {
+    assert(static_cast<bool>(node.get_primitive()->output_data_type) == false &&
+           "Output data type forcing is not supported for count_nonzero_node!");
+    return layout{cldnn::data_types::i32, cldnn::format::bfyx, tensor{1,1,1,4}};
+}
+
+std::string count_nonzero_inst::to_string(count_nonzero_node const& node) {
+    auto desc = node.get_primitive();
+    auto node_info = node.desc_to_json();
+    auto& input = node.input();
+
+    std::stringstream primitive_description;
+
+    json_composite count_nonzero_info;
+    count_nonzero_info.add("input id", input.id());
+    count_nonzero_info.add("output shape", tensor{1,1,1,4});
+
+    node_info->add("count_nonzero info", count_nonzero_info);
+    node_info->dump(primitive_description);
+
+    return primitive_description.str();
+}
+
+count_nonzero_inst::typed_primitive_inst(network& network, count_nonzero_node const& node) : parent(network, node) {}
+
+static std::vector<int64_t> read_vector(cldnn::memory::ptr mem, cldnn::stream& stream) {
+    switch (mem->get_layout().data_type) {
+        case data_types::i32: {
+            mem_lock<int32_t, mem_lock_type::read> lock{mem, stream};
+            return std::vector<int64_t>(lock.begin(), lock.end());
+        }
+        case data_types::i64: {
+            mem_lock<int64_t, mem_lock_type::read> lock{mem, stream};
+            return std::vector<int64_t>(lock.begin(), lock.end());
+        }
+        default: IE_THROW() << "read_vector: unsupported data type";
+    }
+}
+
+// -----------------------------------------------
+// gather_nonzero
+// -----------------------------------------------
+primitive_type_id gather_nonzero::type_id() {
+    static primitive_type_base<gather_nonzero> instance;
+    return &instance;
+}
+
+layout gather_nonzero_inst::calc_output_layout(gather_nonzero_node const& node) {
+    assert(static_cast<bool>(node.get_primitive()->output_data_type) == false &&
+           "Output data type forcing is not supported for gather_nonzero_node!");
+    auto prim = node.get_primitive();
+    return layout{cldnn::data_types::i32, cldnn::format::bfyx, prim->output_shape};
```

Comment:
I think it should be calculated based on the count_nonzero's output shape data

Reply:
The `output_shape` is the output of the previous count_nonzero layer.

The initial value of output_shape is intentionally set to have an undefined value. This is because the entire network is recognized as dynamic only by doing this.

Reply:
Then, isnt it possible to set this gather_non_zero's initial output shape is to be set as undefined too?

Reply:
If you mean the layout `bfyx`, it is fine. Because the number of dimensions of gather_nonzero's output is always two, we can fix the layout of gather_nonzero as `bfyx`.

Reply:
Oh the original code is changed. No problem now.